src/go/k8s/controllers/redpanda/redpanda_node_pvc_controller.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package redpanda
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/component-helpers/storage/volume"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+)
+
+// +kubebuilder:rbac:groups=cluster.redpanda.com,namespace=default,resources=redpandas,verbs=get;list;watch;
+// +kubebuilder:rbac:groups="",namespace=default,resources=persistentvolumeclaims,verbs=get;list;update;patch;delete
+// +kubebuilder:rbac:groups="",namespace=default,resources=persistentvolumes,verbs=get;list;update;patch;delete
+
+// RedpandaNodePVCReconciler watches node objects, and sets annotation to PVC to mark them for deletion
+type RedpandaNodePVCReconciler struct {
+	client.Client
+	// Namespace is where the pvc's are we are watching for
+	Namespace    string
+	OperatorMode bool
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RedpandaNodePVCReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Node{}).WithEventFilter(DeleteEventFilter).Complete(r)
+}
+
+func (r *RedpandaNodePVCReconciler) Reconcile(c context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctx, done := context.WithCancel(c)
+	defer done()
+
+	start := time.Now()
+	log := ctrl.LoggerFrom(ctx).WithName("RedpandaNodePVCReconciler.Reconcile")
+
+	log.Info(fmt.Sprintf("Node %q was found to be deleted, checking for existing PVCs", req.Name))
+
+	result, err := r.reconcile(ctx, req)
+
+	// Log reconciliation duration
+	durationMsg := fmt.Sprintf("reconciliation finished in %s", time.Since(start).String())
+	if result.RequeueAfter > 0 {
+		durationMsg = fmt.Sprintf("%s, next run in %s", durationMsg, result.RequeueAfter.String())
+	}
+	log.Info(durationMsg)
+
+	return result, err
+}
+
+func (r *RedpandaNodePVCReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithName("RedpandaNodePVCReconciler.reconcile")
+	log.Info(fmt.Sprintf("detected node %s deleted; checking if any PVC should be removed", req.Name))
+
+	redpandaNameList := make([]string, 0)
+	if r.OperatorMode {
+		opts := &client.ListOptions{Namespace: r.Namespace}
+		redpandaList := &v1alpha1.RedpandaList{}
+		if errGET := r.Client.List(ctx, redpandaList, opts); errGET != nil {
```

Comment:
The `errGET` seems wrong when there is List call not Get. Can the `errGET` name be `err`?